exporter/signalfxexporter/config.go

```diff
@@ -24,17 +23,24 @@ import (
 	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/splunk"
 )
 
-const (
-	translationRulesConfigKey = "translation_rules"
-)
+type translationRulesConfig struct {
+	TranslationRules []translation.Rule `mapstructure:"translation_rules"`
+}
 
 var defaultTranslationRules = func() []translation.Rule {
-	cfg, err := loadConfig([]byte(translation.DefaultTranslationRulesYaml))
+	var data map[string]any
```

Comment:
New functionality here is essentially copied from `loadConfig`. `loadConfig` takes YAML and parses it into the main `Config` object, but this function only cares about the now deleted `translation_rules` option, so it can't parse the YAML into a `Config` object anymore.